### PR TITLE
Add tests for all the books and ModelManager

### PR DESCRIPTION
**Pull request by zhixianggg**
Tests, tests and more tests.

### CODE REVIEW

File: src/test/java/seedu/address/model/CustomerBookTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import seedu.address.model.customer.Customer;
+import seedu.address.model.customer.exceptions.DuplicateCustomerException;
+import seedu.address.testutil.CustomerBuilder;
+
+public class CustomerBookTest {
+
+    private final CustomerBook customerBook = new CustomerBook();
+
+    @Test
+    public void constructor() {
```
Comment (uberSaiyan) [inline src/test/java/seedu/address/model/CustomerBookTest.java]: Would suggest changing this to fit the format of <method>_<case>_<expectation> i.e. constructor_noArgument_equalsEmptyList or something like that

File: src/test/java/seedu/address/model/OrderBookTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import javafx.collections.ObservableList;
+import seedu.address.model.order.Order;
+import seedu.address.model.order.exceptions.DuplicateOrderException;
+
+public class OrderBookTest {
+
+    private final OrderBook orderBook = new OrderBook();
+
+    @Test
+    public void constructor() {
```
Comment (uberSaiyan) [inline src/test/java/seedu/address/model/OrderBookTest.java]: Similar to CustomerBookTest

File: src/test/java/seedu/address/model/OrderBookTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    public void resetData_withValidReadOnlyDataBookOrder_replacesData() {
+        OrderBook newData = getTypicalOrderBook();
+        orderBook.resetData(newData);
+        assertEquals(newData, orderBook);
+    }
+
+    @Test
+    public void resetData_withDuplicateOrders_throwsDuplicateOrderException() {
+        // Two orders with the same identity fields
+        Order editedVipOrder = (Order) VIPORDER.clone();
```
Comment (uberSaiyan) [inline src/test/java/seedu/address/model/OrderBookTest.java]: Clone is no longer supported

File: src/test/java/seedu/address/model/OrderBookTest.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    @Test
+    public void hasOrder_orderInOrderBook_returnsTrue() {
+        orderBook.addOrder(VIPORDER);
+        assertTrue(orderBook.hasOrder(VIPORDER));
+    }
+
+    @Test
+    public void hasOrder_orderWithSameIdentityFieldsInOrderBook_returnsTrue() {
+        orderBook.addOrder(VIPORDER);
+        Order clonedVipOrder = (Order) VIPORDER.clone();
```
Comment (uberSaiyan) [inline src/test/java/seedu/address/model/OrderBookTest.java]: Similar to above

File: src/test/java/seedu/address/model/PhoneBookTest.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    public void resetData_withValidReadOnlyDataBookPhone_replacesData() {
+        PhoneBook newData = getTypicalPhoneBook();
+        phoneBook.resetData(newData);
+        assertEquals(newData, phoneBook);
+    }
+
+    @Test
+    public void resetData_withDuplicatePhones_throwsDuplicatePhoneException() {
+        // Two phones with the same id
+        Phone editediPhoneXr = new PhoneBuilder(IPHONEXR, true)
```
Comment (uberSaiyan) [inline src/test/java/seedu/address/model/PhoneBookTest.java]: PhoneBuilder constructor will be changed too, >.<

File: src/test/java/seedu/address/testutil/OrderBuilder.java [hunk 6]
```diff
@@ -42,7 +46,22 @@ public OrderBuilder() {
...
+        status = orderToCopy.getStatus();
+        schedule = orderToCopy.getSchedule();
+        tags = new HashSet<>(orderToCopy.getTags());
+    }
+
     public OrderBuilder(Order orderToCopy) {
+        id = UUID.randomUUID();
         customer = orderToCopy.getCustomer();
         phone = orderToCopy.getPhone();
         price = orderToCopy.getPrice();
```
Comment (uberSaiyan) [inline src/test/java/seedu/address/testutil/OrderBuilder.java]: This gonna have merge conflicts

File: src/test/java/seedu/address/model/ModelManagerTest.java [hunk 7]
```diff
@@ -114,19 +215,21 @@ public void equals() {
         assertFalse(modelManager.equals(5));
 
         // different addressBook -> returns false
```
Comment (qiujiaaa) [inline src/test/java/seedu/address/model/ModelManagerTest.java]: different *customerBook
Not sure if we should have 3 more test for each different book.

File: src/test/java/seedu/address/model/OrderBookTest.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        assertThrows(DuplicateOrderException.class, () -> orderBook.resetData(newData));
+    }
+
+    @Test
+    public void hasOrder_nullOrder_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> orderBook.hasOrder(null));
+    }
+
+    @Test
+    public void hasOrder_orderNotInAddressBook_returnsFalse() {
```
Comment (qiujiaaa) [inline src/test/java/seedu/address/model/OrderBookTest.java]: *orderNotInOrderBook

File: src/test/java/seedu/address/model/PhoneBookTest.java [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+    @Test
+    public void hasPhone_nullPhone_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> phoneBook.hasPhone(null));
+    }
+
+    @Test
+    public void hasPhone_phoneNotInPhoneBook_returnsFalse() {
+        assertFalse(phoneBook.hasPhone(IPHONEXR));
+    }
```
Comment (qiujiaaa) [inline src/test/java/seedu/address/model/PhoneBookTest.java]: should we add hasPhone_phoneInPhoneBook_returnsTrue() in?

**Pull request merged**